### PR TITLE
Adding Pre and Post padding functionalities to pad_sequence function

### LINKED ISSUE #10536: [feature request] Adding Pre and Post padding functionalities to pad_sequence function
I would like to add **pre** and **post** padding functionalities to ````torch.nn.utils.rnn.pad_sequence```` .
These functionalities haven't been included in the [function](https://pytorch.org/docs/stable/_modules/torch/nn/utils/rnn.html#pad_sequence)

It would be greatly beneficial for people using seq models like LSTM to have sequences created with both **pre** and **post** padding the sequences passed into the args. Other libraries like keras offer these functionalities in their APIs like: ````data = pad_sequences(data, maxlen=50, dtype='float32', padding='pre')````

Besides this, we can add another functionality to allow an argument(with optional value) to know the length of the sequences to be padded to. This is currently done with length of ````sequence[0]````(sequences passed in decreasing order of length).

Thanks!

cc @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by panzhufeng**
Fixes #10536 
For the function `torch.nn.utils.rnn.pad_sequence` :
* Add another argument `padding_type`(string, 'pre' or 'post') to indicate padding either before or after each sequence

### CODE REVIEW

**Comment by vishwakftw** [comment]
cc @zou3519 , who is porting `pad_sequence` to ATen.

File: torch/nn/utils/rnn.py [hunk 1]
```diff
@@ -320,13 +321,22 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
...
+                out_tensor[:length, i, ...] = tensor
+    elif padding_type == 'pre':
+        for i, tensor in enumerate(sequences):
+            length = tensor.size(0)
+            if batch_first:
+                out_tensor[i, -length:, ...] = tensor
+            else:
+                out_tensor[-length:, i, ...] = tensor
+    else:
+        raise ValueError("Padding_type must be 'post' or 'pre'")
```
Comment (zou3519) [inline torch/nn/utils/rnn.py]: Nit: say what padding_type the user passed in:
`("pad_sequence: padding_type must be  'post' or 'pre', got: {}".format(padding_type)`

**Review by zou3519 — COMMENTED** [review]
lgtm, but please add a test for the pre-padding type @vufg.

**Comment by zou3519** [comment]
btw, I am indeed moving pad_sequence to aten (#10555). Your changes look good so I think we should merge your PR first @vufg and then I can port those changes to aten.

**Comment by vufg** [comment]
@zou3519 I added the test accordingly and fixed some check error. Thanks for your review in advance!

**Comment by fmassa** [comment]
ping @zou3519

**Comment by vufg** [comment]
@zou3519  Do you have any idea who should I @ for the code review? Thanks

File: torch/nn/utils/rnn.py [hunk 2]
```diff
@@ -318,15 +319,26 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
...
-    for i, tensor in enumerate(sequences):
-        length = tensor.size(0)
-        # use index notation to prevent duplicate references to the tensor
-        if batch_first:
-            out_tensor[i, :length, ...] = tensor
-        else:
-            out_tensor[:length, i, ...] = tensor
+
+    if not isinstance(padding_type, str):
+        raise TypeError("Expected padding_type to be str type but {} found".format(type(padding_type)))
```
Comment (apaszke) [inline torch/nn/utils/rnn.py]: I think this is unnecessary. There's an error case below for other values too

**Review by apaszke — APPROVED** [review]

File: torch/nn/utils/rnn.py (hunk 2 above)
Comment (vufg) [reply torch/nn/utils/rnn.py]: Oop, fixed
